app/test/unit/git/stash-test.ts

```diff
@@ -0,0 +1,32 @@
+import * as FSE from 'fs-extra'
+import * as path from 'path'
+import { Repository } from '../../../src/models/repository'
+import { setupEmptyRepository } from '../../helpers/repositories'
+import { GitProcess } from 'dugite'
+import { MagicStashString, getStashEntries } from '../../../src/lib/git/stash'
+
+describe('git/stash', () => {
+  describe('getStashEntries', () => {
+    it('returns all stash entries created by desktop', async () => {
+      const repository = await setupEmptyRepository()
+
+      const readme = path.join(repository.path, 'README.md')
+      await FSE.writeFile(readme, '')
+      await GitProcess.exec(['add', 'README.md'], repository.path)
+      await GitProcess.exec(['commit', '-m', 'initial commit'], repository.path)
+      await FSE.appendFile(readme, 'Hello\n')
+      await stash(repository)
+
+      const stashEntries = await getStashEntries(repository)
+
+      expect(stashEntries).toHaveLength(1)
```

Comment:
Is it worth also doing a check here that `stashEntries[0]` matches some expected text? Should it be the current branch?

Reply:
I'm asserting the branch name now, but I've decided to not assert the stash Id (at least for the time being) because I'll be adding another function in another PR that can return the SHA of a stash entry.